openedx/core/djangoapps/content_libraries/api/blocks.py

```diff
@@ -321,7 +336,9 @@ def validate_can_add_block_to_library(
     # Make sure the proposed ID will be valid:
     validate_unicode_slug(block_id)
     # Ensure the XBlock type is valid and installed:
-    XBlock.load_class(block_type)  # Will raise an exception if invalid
+    block_class = XBlock.load_class(block_type)  # Will raise an exception if invalid
+    if block_class.has_children:
+        raise IncompatibleTypesError("XBlocks with children are not supported in content libraries")
```

Comment:
Please add the block type and block_id in the error being raised, so that it's easier to pinpoint the problematic content.